### DISCUSSION TITLE
In memory event store does not perform consistency check

### DISCUSSION THREAD

**Discussion by midnight-blue** [modernice/goes#183] [Q&A] [answered]
Hello,

I was checking the [Event Store](https://goes.modernice.dev/guide/events/event-store.html) documentation where the consistency checks for aggregate events are mentioned.

However, when I was trying out the exact code from the documentation i got a passing test when a failure was expected:

```
func TestAggregateConsistency(t *testing.T) {
	store := eventstore.New()

	id := uuid.New()
	name := "example"
	version := 1

	// We have two events with the same Aggregate Version.
	events := []event.Event{
		event.New("foo", "foo-data", event.Aggregate(id, name, version)).Any(),
		event.New("bar", "bar-data", event.Aggregate(id, name, version)).Any(),
	}

	// This will fail due to the consistency check.
	if err := store.Insert(context.TODO(), events...); err != nil {
		panic(fmt.Errorf("insert events: %w", err))

	}
}

```

is passing even though its supposed to fail. 
Am I doing something wrong or misunderstanding something here?

Thanks in advance,
Martin

**Comment by bounoable** [comment]
Hello @midnight-blue. Seems like I missed this when implementing the event store, as the in-memory store should only be used during development/testing and I didn't expect this to be a "required" feature for development purposes. But you're right, the event store should validate the event versions.

Is this something that is currently blocking your development? Because I won't get to fix this until at least January.

**Reply by MartinSchwarzmaier** [reply, marked as answer]
Hey @bounoable ,
Thanks for the quick reply! 
No, of course this is not important in any way. I was simply wondering if I misunderstood something as I'm building up my understanding of all the concepts involved. Please keep up the awesome work!
